### PR TITLE
Enable pluggable LiteMap backends

### PR DESCRIPTION
**Pull request by sffc**
This is cleanup I've kind-of wanted to do for a while. It's also aspirationally motivated by #1767.

### CODE REVIEW

**Comment by sffc** [comment]
Docs preview:

https://icu4x-pr-artifacts.storage.googleapis.com/gha/40f01a2db59790a60a4035a1ef3197c227550555/docs/litemap/index.html

I need to fix up:

1. Export `LiteMapWithStore`
2. Add crate-level docs
3. Generalize the Serde impl?

File: utils/litemap/src/lib.rs [hunk 1]
```diff
@@ -34,8 +34,23 @@ extern crate std;
...
 
+use alloc::vec::Vec;
+use core::marker::PhantomData;
+
 mod map;
 #[cfg(feature = "serde")]
 mod serde;
+pub mod store;
+
+pub type LiteMap<K, V> = map::LiteMapWithStore<K, V, Vec<(K, V)>>;
```
Comment (Manishearth) [inline utils/litemap/src/lib.rs:45]: thought: should we be using a defaulted type parameter instead?

**Review by Manishearth — DISMISSED** [review]
praise: quite cleanly done!

File: utils/litemap/src/lib.rs (hunk 1 above)
Comment (sffc) [reply utils/litemap/src/lib.rs]: Done.

**Review by Manishearth — DISMISSED** [review]

**Review by Manishearth — DISMISSED** [review]

**Review by Manishearth — APPROVED** [review]

**Pull request merged**